packages/gatsby-recipes/src/renderer/render.js

```diff
@@ -171,14 +152,43 @@ const handleResource = (resourceName, context, props) => {
   const inFlightPromise = inFlightCache.get(cacheKey)
 
   if (cachedResult) {
-    updateResource(cachedResult)
     return cachedResult
   }
 
   if (inFlightPromise) {
     throw inFlightPromise
   }
 
+  // If this resource requires another resource to be created before it,
+  // check to see if they're created. The first time this is called,
+  // create a promise which we cache and keep throwing until
+  // all the dependencies are created & then we reject the promise
+  // to trigger andother render where we finally can create the
+  // now no-longer-blocked resource.
+  //
+  // TODO test this when we can mock resources by varying what
+  // resources depend on what & which return first and ensuring
+  // resources end in right order.
+  if (mode === `apply` && resourcePlan.dependsOn) {
```

Comment:
Since `resourcePlan` was defined as `const resourcePlan = allResources?.find(` it might be `undefined` (so a prop access would throw). Is there an implicit reason why it would never be for the case of `mode === 'apply'`?

Reply:
we check `apply` as the blocking on uncreated dependencies only happens during the apply (install) step.

We'll always find a `resourcePlan` as `handleResource` is called directly from the list of all resources — it's a bit annoying but for reasons, we just have to restitch up different data streams.